enh(netapp::ontap::snmp): added uptime packaging, fixed Disk-Failed command, enhanced documenation page (#3906)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/hardware-storage-netapp-ontap-snmp.md

**Configuration > Monitoring Connector Manager** menu:
* [Base Pack](./base-generic.md)

## Pack assets

### Templates

The Monitoring Connector **NetApp Ontap SNMP** brings a host template:

* **HW-Storage-NetApp-Ontap-SNMP-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="HW-Storage-NetApp-Ontap-SNMP-custom" label="HW-Storage-NetApp-Ontap-SNMP-custom">

| Service Alias | Service Template                                  | Service Description                                                      |
|:--------------|:--------------------------------------------------|:-------------------------------------------------------------------------|
| Cache-Age     | HW-Storage-NetApp-Ontap-Cache-Age-SNMP-custom     | Check age in minutes of the oldest read-only blocks in the buffer cache. |
| Cp-Statistics | HW-Storage-NetApp-Ontap-Cp-Statistics-SNMP-custom | Check consistency point metrics.                                         |
| Cpu-Load      | HW-Storage-NetApp-Ontap-Cpu-Load-SNMP-custom      | Check CPU usage.                                                         |

> | Disk-Failed   | HW-Storage-NetApp-Ontap-Disk-Failed-SNMP-custom   | Check the current number of disk broken                                  |

Reviewer: ```suggestion
| Disk-Failed   | HW-Storage-NetApp-Ontap-Disk-Failed-SNMP-custom   | Check the current number of malfunctioning disks                                  |
```

| Global-status | HW-Storage-NetApp-Ontap-Global-status-SNMP-custom | Check the overall status of the  appliance                               |

> | Nvram         | HW-Storage-NetApp-Ontap-Nvram-SNMP-custom         | Check current status of the NVRAM batteries                              |

Reviewer: ```suggestion
| Nvram         | HW-Storage-NetApp-Ontap-Nvram-SNMP-custom         | Check the current status of the NVRAM batteries                              |
```

| Shelf         | HW-Storage-NetApp-Ontap-Shelf-SNMP-custom         | Check Shelves hardware                                                   |

> The services listed above are created automatically when the **HW-Storage-NetApp-Ontap-SNMP-custom** host template is used.

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias          | Service Template                                           | Service Description                                                                   | Discovery |
|:-----------------------|:-----------------------------------------------------------|:--------------------------------------------------------------------------------------|:---------:|
| Aggregates             | HW-Storage-NetApp-Ontap-Aggregates-SNMP-custom             | Check state of one or several aggregates                                              |           |
| Cluster-Nodes          | HW-Storage-NetApp-Ontap-Cluster-Nodes-SNMP-custom          | Check cluster nodes                                                                   |     X     |

> | Fan                    | HW-Storage-NetApp-Ontap-Fan-SNMP-custom                    | Check if fans are failed                                                              |           |

Reviewer: ```suggestion
| Fan                    | HW-Storage-NetApp-Ontap-Fan-SNMP-custom                    | Check if fans have failed                                                              |           |
```

| File-System-Global     | HW-Storage-NetApp-Ontap-File-System-Global-SNMP-custom     | Check filesystem usage                                                                |     X     |

> | Ndmpsessions           | HW-Storage-NetApp-Ontap-Ndmpsessions-SNMP-custom           | Check current total of ndmp sessions opened                                           |           |

Reviewer: ```suggestion
| Ndmpsessions           | HW-Storage-NetApp-Ontap-Ndmpsessions-SNMP-custom           | Check current total of open NDMP sessions                                           |           |
```

| Partner-Status         | HW-Storage-NetApp-Ontap-Partner-Status-SNMP-custom         | Check status of clustered failover partner                                            |           |
| Plexes                 | HW-Storage-NetApp-Ontap-Plexes-SNMP-custom                 | Check plexes                                                                          |     X     |

> | Psu                    | HW-Storage-NetApp-Ontap-Psu-SNMP-custom                    | Check if power supplies are failed                                                    |           |

Reviewer: ```suggestion
| Psu                    | HW-Storage-NetApp-Ontap-Psu-SNMP-custom                    | Check if power supplies have failed                                                    |           |
```

| Quotas                 | HW-Storage-NetApp-Ontap-Quotas-SNMP-custom                 | Check quotas                                                                          |           |

> | Share-Calls            | HW-Storage-NetApp-Ontap-Share-Calls-SNMP-custom            | Check cifs and nfs calls per seconds.                                                 |           |

Reviewer: ```suggestion
| Share-Calls            | HW-Storage-NetApp-Ontap-Share-Calls-SNMP-custom            | Check CIFS and NFS calls per second.                                                 |           |
```

> | Snapshot-Age-Global    | HW-Storage-NetApp-Ontap-Snapshot-Age-Global-SNMP-custom    | Check snapshot age of volumes                                                         |           |

Reviewer: ```suggestion
| Snapshot-Age-Global    | HW-Storage-NetApp-Ontap-Snapshot-Age-Global-SNMP-custom    | Check the age of volume snapshots |           |
```

> | Snapshot-Age-Name      | HW-Storage-NetApp-Ontap-Snapshot-Age-Name-SNMP-custom      | Check snapshot age of volumes                                                         |           |

Reviewer: ```suggestion
| Snapshot-Age-Name      | HW-Storage-NetApp-Ontap-Snapshot-Age-Name-SNMP-custom      | Check the age of volume snapshots                                                         |           |
```

|:----------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| SNMP Agents     | Discover your resources through an SNMP subnet scan. You need to install the [Generic SNMP](./applications-protocol-snmp.md) connector to get the discovery rule and create a template mapper for the **HW-Storage-NetApp-Ontap-SNMP-custom** host template. |

More information about discovering hosts automatically is available on the [dedicated page](/docs/monitoring/discovery/hosts-discovery).

#### Service discovery

| Rule name                               | Description                             |
|:----------------------------------------|:----------------------------------------|
| Net-Netapp-Ontap-SNMP-Cluster-Node-Name | Discover cluster nodes and monitor them |

> | Net-Netapp-Ontap-SNMP-Disk-Name         | Discover disk broken and monitor them   |

Reviewer: ```suggestion
| Net-Netapp-Ontap-SNMP-Disk-Name         | Discover malfunctioning disks and monitor them   |
```

| count_psu                     | count |
| count_raid                    | count |
| count_temperature             | count |
| count_voltage                 | count |
| speed_*channel-shelf-address* | rpm   |
| temp_*channel-shelf-address*  | C     |
| volt_*channel-shelf-address*  | mV    |

</TabItem>
<TabItem value="Snapshot-Age-Global" label="Snapshot-Age-Global">

| Name      | Unit  |
|:----------|:------|
| snapshots | N/A   |

</TabItem>
<TabItem value="Snapshot-Age-Name" label="Snapshot-Age-Name">

| Name      | Unit  |
|:----------|:------|
| snapshots | N/A   |

</TabItem>
<TabItem value="Temperature" label="Temperature">

No metrics for this service.

</TabItem>
<TabItem value="Volume-Options-Generic" label="Volume-Options-Generic">

| Name    | Unit  |
|:--------|:------|
| status  | N/A   |
| status  | N/A   |
| options | N/A   |
| options | N/A   |
| failed  | N/A   |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
</Tabs>

## Prerequisites

### Equipment specificity

If your netapp storage is in 'c-mode', following services won't work:
- Global-Status
- Share-Calls
- Cache-Age
- Ndmpsessions

> 

Author: ```suggestion

> The current plugin is not fully compatible with device versions higher than 8.2.5. You can vote for its improvement in this [TheWatch idea](https://thewatch.centreon.com/ideas/mise-a-jour-plugin-pack-netapp-ontap-snmp-3571).

```
